adev/shared-docs/pipeline/guides/extensions/docs-code/docs-code.ts

```diff
@@ -18,7 +18,7 @@ export interface DocsCodeToken extends CodeToken {
 
 // Capture group 1: all attributes on the opening tag
 // Capture group 2: all content between the open and close tags
-const singleFileSelfClosingCodeRule = /^\s*<docs-code\s([^>]*)((?:.(?!\/>))*)\/>/s;
+const singleFileSelfClosingCodeRule = /^\s*<docs-code\s*(.*?)\s*(?=\/>)\/>\s*$/;
 const singleFileCodeRule = /^\s*<docs-code\s([^>]*)>((?:.(?!\/docs-code))*)<\/docs-code>/s;
```

Comment:
If one can make certain assumptions about the code snippets (for example, that attributes values are always going to be wrapped in double quotes), then I would suggst replacing the two regular expressions with the following one:

```suggestion
const singleFileCodeRule = /^\s*<docs-code((?:\s+[\w-]+(?:="[^"]*")?)*)\s*(?:\/>|(.*?)<\/docs-code>)/s
```

This will be able to match both self-closing and regular `<docs-code>` elements. The first capturing group will have the attributes and the second capturing group will have the element contents (for regular elements) and will be `undefined` for self-closing ones.

If one needs to also support single quotes for attribute values (e.g. `<docs-code attr='value' />`) or even no quotes (e.g. `<docs-code attr=value />`), then the regex could be changed to:

```suggestion
const singleFileCodeRule = /^\s*<docs-code((?:\s+[\w-]+(?:="[^"]*"|='[^']*'|=\S*)?)*)\s*(?:\/>|(.*?)<\/docs-code>)/s
```

Reply:
Wow this is really close to the expected output 

now only problem is with `/^\s*<docs-code((?:\s+[\w-]+(?:="[^"]*"|='[^']*'|=\S*)?)*)\s*(?:\/>|(.*?)<\/docs-code>)/s`
for below input 
```
<docs-code language="html">
<element i18n="{i18n_metadata}">{string_to_translate}</element>
</docs-code>
```

the O/P is 
```
>
<element i18n="{i18n_metadata}">{string_to_translate}</element>
```

<img width="1215" alt="Screenshot 2024-08-04 at 21 26 00" src="https://github.com/user-attachments/assets/15ad2d67-01cd-4c34-a011-5721f8f102a7">

Reply:
I made some changes and it worked